staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -327,7 +342,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			case err == io.ErrUnexpectedEOF:
 				klog.V(1).Infof("%s: Watch for %v closed with unexpected EOF: %v", r.name, r.expectedTypeName, err)
 			default:
-				utilruntime.HandleError(fmt.Errorf("%s: Failed to watch %v: %v", r.name, r.expectedTypeName, err))
+				errToReturn = fmt.Errorf("Failed to watch %v: %v", r.expectedTypeName, err)
```

Comment:
Why not return the error right away?

Reply:
i wanted to preserve the current behavior where we don't return on connection refused (i.e., line 351)